migration/model.go

```diff
@@ -0,0 +1,101 @@
+package migration
+
+import (
+	"encoding/binary"
+
+	"github.com/iov-one/weave"
+	"github.com/iov-one/weave/errors"
+	"github.com/iov-one/weave/orm"
+)
+
+var _ orm.CloneableData = (*Schema)(nil)
+
+func init() {
+	MustRegister(1, &Schema{}, NoModification)
+}
+
+func (s *Schema) Validate() error {
+	if s.Version < 1 {
+		return errors.Wrap(errors.ErrInvalidModel, "version must be greater than zero")
+	}
+	if s.Pkg == "" {
+		return errors.Wrap(errors.ErrInvalidModel, "pkg is required")
+	}
+	return nil
+}
+
+func (s *Schema) Copy() orm.CloneableData {
+	return &Schema{
+		Metadata: s.Metadata.Copy(),
+		Version:  s.Version,
+		Pkg:      s.Pkg,
+	}
+}
+
+// schemaID returns a detereministic ID of this schema instance.
+func schemaID(pkg string, version uint32) []byte {
+	raw := make([]byte, len(pkg)+4)
+	copy(raw, pkg)
+	binary.LittleEndian.PutUint32(raw[len(pkg):], version)
+	return raw
+}
+
+type SchemaBucket struct {
+	orm.Bucket
+}
+
+func NewSchemaBucket() *SchemaBucket {
+	b := orm.NewBucket("schema", orm.NewSimpleObj(nil, &Schema{}))
+	return &SchemaBucket{Bucket: b}
+}
+
+// CurrentSchema returns the current version of the schema for a given package.
+// It returns ErrNotFound if no schema version was registered for this package.
+// Minimum schema version is 1.
+func (b *SchemaBucket) CurrentSchema(db weave.KVStore, packageName string) (uint32, error) {
+	for ver := uint32(1); ver < 10000; ver++ {
```

Comment:
If you use BigEndian (above), you can use a reverse iterator as I mentioned, and grab the first element from it. This is one read and will not have a limit.

Reply:
I was thinking about it. I will leave it for now and come back to this before merging to master.